internal/controllers/cryostat_controller_test.go

```diff
@@ -2053,6 +2075,20 @@ func (t *cryostatTestInput) checkDeploymentHasTemplates() {
 	Expect(volumeMounts).To(Equal(expectedVolumeMounts))
 }
 
+func (t *cryostatTestInput) checkDeploymentHasAuthProperties() {
+	deployment := &appsv1.Deployment{}
+	err := t.Client.Get(context.Background(), types.NamespacedName{Name: "cryostat", Namespace: "default"}, deployment)
+	Expect(err).ToNot(HaveOccurred())
+
+	volumes := deployment.Spec.Template.Spec.Volumes
+	expectedVolumes := test.NewVolumeWithAuthProperties(t.TLS)
+	Expect(volumes).To(ConsistOf(expectedVolumes))
+
+	volumeMounts := deployment.Spec.Template.Spec.Containers[0].VolumeMounts
+	expectedVolumeMounts := test.NewVolumeMountsWithAuthProperties(t.TLS)
+	Expect(volumeMounts).To(ConsistOf(expectedVolumeMounts))
```

Comment:
I ran into issues while comparing slices with Gomega `Equal`  and the elements in two slices are of different orders.

 `ConsistOf` solved the issue. I am thinking of using this for other places that comparing slices (mostly Volume and VolumeClaims).

Reply:
Very good! I think I've used ConsistOf in other places, but it's definitely the right approach for situations like this.